lib/medtronic/medtronicSimulator.js

```diff
@@ -66,6 +77,111 @@ exports.make = function(config){
   }
 
   return {
+    basal: function(event) {
+      ensureTimestamp(event);
+
+      if (currBasal != null) {
+
+        if(currBasal.deliveryType === 'temp' && event.deliveryType === 'temp' && event.duration !== 0) {
+          // temp basal was updated
+          currBasal.with_expectedDuration(currBasal.duration)
+                   .with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          event.suppressed = _.clone(currBasal.suppressed);
```

Comment:
Indeed. When a _percent_ temp basal is running, the "Set/edit" option doesn't do anything. With a _rate_ temp basal, you can edit both the duration and the rate itself, after which it stops the current temp basal and starts a new one.